objectbox/lib/src/native/query/query.dart

```diff
@@ -631,7 +632,17 @@ class Query<T> {
   }
 
   Query._(this._store, Pointer<OBX_query_builder> cBuilder, this._entity)
-      : _cQuery = checkObxPtr(C.query(cBuilder), 'create query');
+      : _cQuery = checkObxPtr(C.query(cBuilder), 'create query') {
+    initializeDartAPI();
+
+    // Keep the finalizer so we can detach it when close() is called manually.
+    _cFinalizer =
+        C.dartc_attach_finalizer(this, native_query_close, _cQuery.cast(), 256);
+    if (_cFinalizer == nullptr) {
+      close();
+      throwLatestNativeError();
```

Comment:
Just checking: can `C.query_close` work if `C.dartc_attach_finalizer` failed? E.g. if not, would always replace the original error.

Reply:
yep, it should work as just fine.